pennylane/vqe/vqe.py

```diff
@@ -174,35 +191,39 @@ def simplify(self):
           (-1) [X0]
         + (1) [Y2]
         """
-        coeffs = []
+        data = []
         ops = []
 
-        for c, op in zip(self.coeffs, self.ops):
+        for i in range(len(self.ops)):  # pylint: disable=consider-using-enumerate
+            op = self.ops[i]
+            c = self.coeffs[i]
             op = op if isinstance(op, Tensor) else Tensor(op)
```

Comment:
I am not sure what to do here. Parsing to Tensor queues the new tensor, even if it is never used except from the comparison below. I tried to a) add a `do_queue` argument to the tensor class, but then all sorts of things started failing; and b) to get the active tape and stop recording the simplify function, but then the Tensors that are used as the new observables are not in the queue if `simplify` is called after the Hamiltonian was created. It is really tricky :) 

Any ideas?

Reply:
You could do
```python
op = op if isinstance(op, Tensor) else qml.transforms.invisible(Tensor)(op)
```

This is simply a shortcut for
```python
active_tape = qml.get_active_tape()

if active_tape is not None:
    with active_tape.stop_recording():
        op = op if isinstance(op, Tensor) else Tensor(op)
else:
    op = op if isinstance(op, Tensor) else Tensor(op)
```

Reply:
Nice, I didn't know this works for a single op as input!

Unfortunately this also may not work, since `op` will sometimes be needed in the queue, since it is added to the list of ops...

I leave it as is, and convinced myself that a cluttered queue is not an urgent problem anywhere...